### PR TITLE
mz enable region: Fix pg_isready and other issues

### PR DESCRIPTION
**Pull request by antifuchs**
This PR adjusts a few things about `mz enable region`'s behavior when waiting for the region to turn ready:

* Split the loading spinner in two: One spinner for enabling the region, one for checking that it's online.
* Use the `-d` command line option to tell pg_isready that this is a connection string.
* Only run pg_isready only after the environment is resolvable, to avoid poisoning the client's DNS cache (just like we do in the webface)

### Motivation

  * This PR fixes a previously unreported bug.

@philip-stoev found out that we're invoking pg_ishealthy with the wrong CLI args, which makes the tool spin forever while the region is already up; so now we're not doing that anymore & `mz enable region` correctly finishes when the region comes up.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/mz/src/api.rs [hunk 1]
```diff
@@ -70,6 +70,7 @@ pub struct Region {
 pub struct Environment {
     pub environmentd_pgwire_address: String,
     pub environmentd_https_address: String,
+    pub resolvable: bool,
```
Comment (mjibson) [inline src/mz/src/api.rs:73]: Does this get populated by our backend? (I'm not familiar with the mz codebase.)

**Review by mjibson — APPROVED** [review]

File: src/mz/src/api.rs (hunk 1 above)
Comment (antifuchs) [reply src/mz/src/api.rs:73]: Yep, it is - the environment controller does a dns resolution of the name against all the authoritative NS servers & sets that to true once they all can resolve the name.

**Comment by philip-stoev** [comment]
@antifuchs can you do a git commit --amend ; git push --force to get the CLA agent unstuck on this PR?

**Comment by antifuchs** [comment]
OOOps, yes!

**Pull request merged**